### PR TITLE
B 3/create entity user

### PR DESCRIPTION
**Pull request by Eju97**
Added the following controllers:

- Get All Users
- Get User by ID
- Create User
- Update User
- Delete User

Added User schema (appointment field still in progress).

Users route added to the API:

- .get('/')
- .get('/:id')
- .post('/')
- .delete('/:id),
- .put('/:id')

### CODE REVIEW

File: src/controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        const result = await Users.findByIdAndUpdate(id, {
+            isActive: false,
+        });
+        if(!result) {
+            throw new Error({
+                message: 'User not found',
+                status: 404,
+            });
+        }
+        return res.status(204).json({
```
Comment (DamianPalavecino) [inline src/controllers/user.js]:
```suggestion
        return res.status(200).json({
```

File: src/controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
+import Users from '../models/User.js';
+import { isValidObjectId } from 'mongoose';
+
+export const getUsers = async (req, res) => {
+    try{
+        const users = await Users.find(req.query);
+        if(!users) {
+            throw new Error({
+                message: 'Users not found',
+                status: 404,
```
Comment (JulianFloresDev) [inline src/controllers/user.js]: change this throw error for:

`return res.status(404).json({
message: 'Users not found',
error: true,
data: undefined
})
`

File: src/controllers/user.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+};
+
+export const getUserById = async (req, res) => {
+    try{
+        const { id } = req.params;
+        if(!isValidObjectId(id)) {
+            throw new Error({
+                message: 'Invalid User ID',
+                status: 400,
+            })
```
Comment (JulianFloresDev) [inline src/controllers/user.js]: Same as above:

`return res.status(404).json({ message: 'Users not found', error: true, data: undefined })`

**Review by DamianPalavecino — APPROVED** [review]
Approved, well done papu

File: src/controllers/user.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    }
+};
+
+export const createUser = async(req, res) => {
+    try{
+        const newUser = new Users({
+            name: req.body.name,
+            lastName: req.body.lastName,
+            phone: req.body.phone,
+            email: req.body.email,
```
Comment (JulianFloresDev) [inline src/controllers/user.js:59]: Add 'isActive' param on `true` when crete new user

**Review by JulianFloresDev — APPROVED** [review]

**Pull request merged**